apps/rush/src/utilities/InstallManager.ts

```diff
@@ -298,19 +302,16 @@ export default class InstallManager {
     sortedRushProjects.sort(
       (a: RushConfigurationProject, b: RushConfigurationProject) => a.tempProjectName.localeCompare(b.tempProjectName)
     );
+
     for (const rushProject of sortedRushProjects) {
       const packageJson: IPackageJson = rushProject.packageJson;
 
-      // Example: "C:\MyRepo\common\temp\projects\my-project-2"
-      const tempProjectFolder: string = path.dirname(rushProject.tempPackageJsonFilename);
-      fsx.mkdirsSync(tempProjectFolder);
+      // Example: "my-project-2"
+      const unscopedTempProjectName: string = path.basename(rushProject.tempPackageTarballFilename, '.tgz');
```

Comment:
>path.basename(rushProject.tempPackageTarballFilename, '.tgz'); [](start = 46, length = 62)

This is a weird design.  On line 393 you use "path.dirname(rushProject.tempPackageTarballFilename)" to calculate the same thing.

Why not just make "unscopedTempProjectName" your API property (e.g. called "tempProjectUniqueName"), and then construct the other derivatives from that.  I'd argue that these derivatives are all internal guts of InstallManager, not something that rush-lib consumers should be privy to.

installCommonModules() is a slightly tricky case, but you could just make a private helper function getTarballPath(project)